### ISSUE TITLE
acli new fails to install BLT correctly and errors out

### ISSUE THREAD

**Issue by anavarre** [acquia/cli#427]
Ensured the project directory was empty before running the command.

<img width="1001" alt="blt-error" src="https://user-images.githubusercontent.com/293478/109015227-311bdd80-76b5-11eb-8353-aa5d128d93f4.png">

Full trace:

```
ide:~/project $ ls -l
total 0
ide:~/project $ cd ..
ide:~ $ acli new /home/ide/project/ -vvv

Box Requirements Checker
========================

> Using PHP 7.4.14
> PHP is using the following php.ini file:
  /usr/local/php7.4/etc/cli/php.ini

> Checking Box requirements:
  ✔ The application requires the version "^7.3 | ^8.0" or greater.
  ✔ The application requires the extension "json".
  ✔ The package "brick/math" requires the extension "json".
  ✔ The package "guzzlehttp/guzzle" requires the extension "json".
  ✔ The package "ramsey/uuid" requires the extension "json".
  ✔ The package "zumba/amplitude-php" requires the extension "curl".
  
                                                                                
 [OK] Your system is ready to run the application.                              
                                                                                

Acquia CLI version: 1.5.0
Acquia recommends most customers use acquia/drupal-recommended-project, which includes useful utilities such as Acquia Connector.
acquia/drupal-minimal-project is the most minimal application that will run on the Cloud Platform.

 Choose a starting project:
  [0] acquia/drupal-recommended-project
  [1] acquia/drupal-minimal-project
 > 0

Creating project. This may take a few minutes
Creating a "acquia/drupal-recommended-project" project at "./project"
Installing acquia/drupal-recommended-project (v1.1.0)
  - Installing acquia/drupal-recommended-project (v1.1.0): Extracting archive
Created project in /home/ide/project
[notice] Command: 'composer' 'create-project' '--no-install' 'acquia/drupal-recommended-project' '/home/ide/project' [Exit: 0]
Loading composer repositories with package information
Updating dependencies
Lock file operations: 185 installs, 0 updates, 0 removals
  - Locking acquia/blt (12.7.0)
  - Locking acquia/drupal-environment-detector (v1.3.0)
  - Locking acquia/lightning (5.1.2)
  - Locking asm89/stack-cors (1.3.0)
  - Locking bower-asset/cropper (v2.3.4)
  - Locking bower-asset/jquery (3.5.1)
  - Locking bower-asset/slick-carousel (v1.8.1)
  - Locking caxy/php-htmldiff (v0.1.11)
  - Locking chi-teck/drupal-code-generator (1.33.1)
  - Locking composer/ca-bundle (1.2.9)
  - Locking composer/composer (2.0.11)
  - Locking composer/installers (v1.10.0)
  - Locking composer/semver (3.2.2)
  - Locking composer/spdx-licenses (1.5.5)
  - Locking composer/xdebug-handler (1.4.5)
  - Locking consolidation/annotated-command (4.2.4)
  - Locking consolidation/comments (1.0.2)
  - Locking consolidation/config (1.2.1)
  - Locking consolidation/filter-via-dot-access-data (1.0.0)
  - Locking consolidation/log (2.0.2)
  - Locking consolidation/output-formatters (4.1.2)
  - Locking consolidation/robo (2.2.2)
  - Locking consolidation/self-update (1.2.0)
  - Locking consolidation/site-alias (3.1.0)
  - Locking consolidation/site-process (4.1.0)
  - Locking container-interop/container-interop (1.2.0)
  - Locking cweagans/composer-patches (1.7.0)
  - Locking defuse/php-encryption (v2.2.1)
  - Locking dflydev/dot-access-data (v1.1.0)
  - Locking dnoegel/php-xdg-base-dir (v0.1.1)
  - Locking doctrine/annotations (1.11.1)
  - Locking doctrine/lexer (1.2.1)
  - Locking doctrine/reflection (1.2.2)
  - Locking drupal/acquia_connector (2.0.0-rc4)
  - Locking drupal/acquia_telemetry-acquia_telemetry (1.0.0-alpha5)
  - Locking drupal/autosave_form (1.2.0)
  - Locking drupal/bg_image_formatter (1.14.0)
  - Locking drupal/blazy (2.1.0)
  - Locking drupal/conflict (2.0.0-alpha2)
  - Locking drupal/consumers (1.11.0)
  - Locking drupal/contact_storage (1.1.0)
  - Locking drupal/core (9.1.4)
  - Locking drupal/core-composer-scaffold (9.1.4)
  - Locking drupal/core-recommended (9.1.4)
  - Locking drupal/core_context (1.0.0-beta4)
  - Locking drupal/crop (2.1.0)
  - Locking drupal/ctools (3.4.0)
  - Locking drupal/ctools_block (3.4.0)
  - Locking drupal/diff (1.0.0)
  - Locking drupal/dropzonejs (2.4.0)
  - Locking drupal/embed (1.4.0)
  - Locking drupal/entity_block (1.0.0-beta3)
  - Locking drupal/entity_browser (2.5.0)
  - Locking drupal/entity_browser_block (1.1.0)
  - Locking drupal/entity_embed (1.1.0)
  - Locking drupal/image_widget_crop (2.3.0)
  - Locking drupal/inline_entity_form (1.0.0-rc8)
  - Locking drupal/jquery_ui (1.4.0)
  - Locking drupal/jquery_ui_draggable (1.2.0)
  - Locking drupal/jquery_ui_droppable (1.2.0)
  - Locking drupal/layout_builder_restrictions (2.8.0)
  - Locking drupal/layout_builder_st (1.0.0-alpha2)
  - Locking drupal/layout_builder_styles (1.0.0-beta4)
  - Locking drupal/layout_library (1.0.0-beta2)
  - Locking drupal/lightning_api (4.6.0)
  - Locking drupal/lightning_core (5.5.0)
  - Locking drupal/lightning_layout (2.9.0)
  - Locking drupal/lightning_media (4.6.0)
  - Locking drupal/lightning_workflow (3.16.0)
  - Locking drupal/media_entity_instagram (3.0.4)
  - Locking drupal/media_entity_twitter (2.6.0)
  - Locking drupal/metatag (1.15.0)
  - Locking drupal/moderation_dashboard (1.0.0-beta2)
  - Locking drupal/moderation_sidebar (1.4.0)
  - Locking drupal/mysql56 (1.1.0)
  - Locking drupal/openapi (2.0.0-rc3)
  - Locking drupal/openapi_jsonapi (2.0.0)
  - Locking drupal/openapi_rest (2.0.0-rc2)
  - Locking drupal/openapi_ui (1.0.0-rc3)
  - Locking drupal/openapi_ui_redoc (1.0.0-rc3)
  - Locking drupal/openapi_ui_swagger (1.0.0-rc4)
  - Locking drupal/page_manager (4.0.0-beta6)
  - Locking drupal/panelizer (5.0.0-beta3)
  - Locking drupal/panels (4.6.0)
  - Locking drupal/panels_ipe (4.6.0)
  - Locking drupal/pathauto (1.8.0)
  - Locking drupal/pendo (1.0.0-alpha4)
  - Locking drupal/profile_switcher (1.0.0-alpha5)
  - Locking drupal/redirect (1.6.0)
  - Locking drupal/schemata (1.0.0-beta2)
  - Locking drupal/schemata_json_schema (1.0.0-beta2)
  - Locking drupal/search_api (1.19.0)
  - Locking drupal/simple_gmap (3.0.0)
  - Locking drupal/simple_oauth (4.5.0)
  - Locking drupal/slick (2.2.0)
  - Locking drupal/slick_entityreference (2.0.0)
  - Locking drupal/token (1.9.0)
  - Locking drupal/video_embed_field (2.4.0)
  - Locking drupal/views_infinite_scroll (1.7.0)
  - Locking drupol/phposinfo (1.6.5)
  - Locking drush/drush (10.4.0)
  - Locking egulias/email-validator (2.1.22)
  - Locking enlightn/security-checker (v1.6.1)
  - Locking ezyang/htmlpurifier (v4.13.0)
  - Locking grasmash/expander (1.0.0)
  - Locking grasmash/yaml-cli (2.0.1)
  - Locking grasmash/yaml-expander (1.4.0)
  - Locking guzzlehttp/guzzle (6.5.5)
  - Locking guzzlehttp/promises (1.4.0)
  - Locking guzzlehttp/psr7 (1.7.0)
  - Locking j7mbo/twitter-api-php (1.0.6)
  - Locking justinrainbow/json-schema (5.2.10)
  - Locking laminas/laminas-diactoros (2.5.0)
  - Locking laminas/laminas-escaper (2.7.0)
  - Locking laminas/laminas-feed (2.13.0)
  - Locking laminas/laminas-stdlib (3.3.0)
  - Locking laminas/laminas-zendframework-bridge (1.1.1)
  - Locking lcobucci/jwt (3.4.5)
  - Locking league/container (2.5.0)
  - Locking league/event (2.2.0)
  - Locking league/oauth2-server (7.4.0)
  - Locking masterminds/html5 (2.7.4)
  - Locking mkalkbrenner/php-htmldiff-advanced (0.0.8)
  - Locking nikic/php-parser (v4.10.4)
  - Locking npm-asset/dropzone (5.7.6)
  - Locking oomphinc/composer-installers-extender (2.0.0)
  - Locking paragonie/random_compat (v9.99.100)
  - Locking pear/archive_tar (1.4.12)
  - Locking pear/console_getopt (v1.4.3)
  - Locking pear/pear-core-minimal (v1.10.10)
  - Locking pear/pear_exception (v1.0.1)
  - Locking phpdocumentor/reflection-common (2.2.0)
  - Locking phpdocumentor/reflection-docblock (5.2.2)
  - Locking phpdocumentor/type-resolver (1.4.0)
  - Locking psr/container (1.0.0)
  - Locking psr/http-factory (1.0.1)
  - Locking psr/http-message (1.0.1)
  - Locking psr/log (1.1.3)
  - Locking psy/psysh (v0.10.6)
  - Locking ralouphie/getallheaders (3.0.3)
  - Locking react/promise (v2.8.0)
  - Locking seld/jsonlint (1.8.3)
  - Locking seld/phar-utils (1.1.1)
  - Locking stack/builder (v1.0.6)
  - Locking swagger-api/swagger-ui (v3.43.0)
  - Locking symfony-cmf/routing (2.3.3)
  - Locking symfony/config (v4.4.19)
  - Locking symfony/console (v4.4.16)
  - Locking symfony/debug (v4.4.16)
  - Locking symfony/dependency-injection (v4.4.16)
  - Locking symfony/error-handler (v4.4.16)
  - Locking symfony/event-dispatcher (v4.4.16)
  - Locking symfony/event-dispatcher-contracts (v1.1.9)
  - Locking symfony/filesystem (v4.4.19)
  - Locking symfony/finder (v5.2.3)
  - Locking symfony/http-client-contracts (v2.3.1)
  - Locking symfony/http-foundation (v4.4.16)
  - Locking symfony/http-kernel (v4.4.16)
  - Locking symfony/mime (v5.1.8)
  - Locking symfony/polyfill-ctype (v1.20.0)
  - Locking symfony/polyfill-iconv (v1.20.0)
  - Locking symfony/polyfill-intl-idn (v1.20.0)
  - Locking symfony/polyfill-intl-normalizer (v1.20.0)
  - Locking symfony/polyfill-mbstring (v1.20.0)
  - Locking symfony/polyfill-php72 (v1.22.1)
  - Locking symfony/polyfill-php73 (v1.22.1)
  - Locking symfony/polyfill-php80 (v1.20.0)
  - Locking symfony/process (v4.4.16)
  - Locking symfony/psr-http-message-bridge (v2.0.2)
  - Locking symfony/routing (v4.4.16)
  - Locking symfony/serializer (v4.4.16)
  - Locking symfony/service-contracts (v2.2.0)
  - Locking symfony/translation (v4.4.16)
  - Locking symfony/translation-contracts (v2.3.0)
  - Locking symfony/twig-bridge (v3.4.47)
  - Locking symfony/validator (v4.4.16)
  - Locking symfony/var-dumper (v5.1.8)
  - Locking symfony/yaml (v4.4.16)
  - Locking twig/twig (v2.14.1)
  - Locking typo3/phar-stream-wrapper (v3.1.6)
  - Locking vardot/blazy (1.8.2.1)
  - Locking webflo/drupal-finder (1.2.2)
  - Locking webmozart/assert (1.9.1)
  - Locking webmozart/path-util (2.3.0)
  - Locking zumba/amplitude-php (1.0.2)
Writing lock file
Installing dependencies from lock file (including require-dev)
Package operations: 185 installs, 0 updates, 0 removals
  - Installing composer/installers (v1.10.0): Extracting archive
  - Installing cweagans/composer-patches (1.7.0): Extracting archive
Gathering patches for root package.
No patches supplied.
Gathering patches for dependencies. This might take a minute.
  - Installing drupal/core-composer-scaffold (9.1.4): Extracting archive
  - Installing psr/log (1.1.3): Extracting archive
  - Installing zumba/amplitude-php (1.0.2): Extracting archive
  - Installing symfony/polyfill-ctype (v1.20.0): Extracting archive
  - Installing symfony/yaml (v4.4.16): Extracting archive
  - Installing symfony/polyfill-mbstring (v1.20.0): Extracting archive
  - Installing twig/twig (v2.14.1): Extracting archive
  - Installing symfony/twig-bridge (v3.4.47): Extracting archive
  - Installing psr/container (1.0.0): Extracting archive
  - Installing symfony/service-contracts (v2.2.0): Extracting archive
  - Installing symfony/polyfill-php80 (v1.20.0): Extracting archive
  - Installing symfony/polyfill-php73 (v1.22.1): Extracting archive
  - Installing symfony/console (v4.4.16): Extracting archive
  - Installing symfony/filesystem (v4.4.19): Extracting archive
  - Installing symfony/config (v4.4.19): Extracting archive
  - Installing dflydev/dot-access-data (v1.1.0): Extracting archive
  - Installing grasmash/yaml-expander (1.4.0): Extracting archive
  - Installing grasmash/yaml-cli (2.0.1): Extracting archive
  - Installing symfony/finder (v5.2.3): Extracting archive
  - Installing symfony/polyfill-php72 (v1.22.1): Extracting archive
  - Installing symfony/polyfill-intl-normalizer (v1.20.0): Extracting archive
  - Installing symfony/polyfill-intl-idn (v1.20.0): Extracting archive
  - Installing ralouphie/getallheaders (3.0.3): Extracting archive
  - Installing psr/http-message (1.0.1): Extracting archive
  - Installing guzzlehttp/psr7 (1.7.0): Extracting archive
  - Installing guzzlehttp/promises (1.4.0): Extracting archive
  - Installing guzzlehttp/guzzle (6.5.5): Extracting archive
  - Installing enlightn/security-checker (v1.6.1): Extracting archive
  - Installing webmozart/assert (1.9.1): Extracting archive
  - Installing webmozart/path-util (2.3.0): Extracting archive
  - Installing webflo/drupal-finder (1.2.2): Extracting archive
  - Installing symfony/var-dumper (v5.1.8): Extracting archive
  - Installing symfony/event-dispatcher-contracts (v1.1.9): Extracting archive
  - Installing symfony/event-dispatcher (v4.4.16): Extracting archive
  - Installing nikic/php-parser (v4.10.4): Extracting archive
  - Installing dnoegel/php-xdg-base-dir (v0.1.1): Extracting archive
  - Installing psy/psysh (v0.10.6): Extracting archive
  - Installing container-interop/container-interop (1.2.0): Extracting archive
  - Installing league/container (2.5.0): Extracting archive
  - Installing symfony/process (v4.4.16): Extracting archive
  - Installing grasmash/expander (1.0.0): Extracting archive
  - Installing consolidation/config (1.2.1): Extracting archive
  - Installing consolidation/site-alias (3.1.0): Extracting archive
  - Installing consolidation/site-process (4.1.0): Extracting archive
  - Installing consolidation/self-update (1.2.0): Extracting archive
  - Installing consolidation/output-formatters (4.1.2): Extracting archive
  - Installing consolidation/log (2.0.2): Extracting archive
  - Installing consolidation/annotated-command (4.2.4): Extracting archive
  - Installing consolidation/robo (2.2.2): Extracting archive
  - Installing consolidation/filter-via-dot-access-data (1.0.0): Extracting archive
  - Installing composer/semver (3.2.2): Extracting archive
  - Installing chi-teck/drupal-code-generator (1.33.1): Extracting archive
  - Installing drush/drush (10.4.0): Extracting archive
  - Installing drupol/phposinfo (1.6.5): Extracting archive
  - Installing typo3/phar-stream-wrapper (v3.1.6): Extracting archive
  - Installing symfony/translation-contracts (v2.3.0): Extracting archive
  - Installing symfony/validator (v4.4.16): Extracting archive
  - Installing symfony/translation (v4.4.16): Extracting archive
  - Installing symfony/serializer (v4.4.16): Extracting archive
  - Installing symfony/routing (v4.4.16): Extracting archive
  - Installing symfony/mime (v5.1.8): Extracting archive
  - Installing symfony/http-foundation (v4.4.16): Extracting archive
  - Installing symfony/psr-http-message-bridge (v2.0.2): Extracting archive
  - Installing symfony/polyfill-iconv (v1.20.0): Extracting archive
  - Installing symfony/http-client-contracts (v2.3.1): Extracting archive
  - Installing symfony/debug (v4.4.16): Extracting archive
  - Installing symfony/error-handler (v4.4.16): Extracting archive
  - Installing symfony/http-kernel (v4.4.16): Extracting archive
  - Installing symfony/dependency-injection (v4.4.16): Extracting archive
  - Installing symfony-cmf/routing (2.3.3): Extracting archive
  - Installing stack/builder (v1.0.6): Extracting archive
  - Installing pear/pear_exception (v1.0.1): Extracting archive
  - Installing pear/console_getopt (v1.4.3): Extracting archive
  - Installing pear/pear-core-minimal (v1.10.10): Extracting archive
  - Installing pear/archive_tar (1.4.12): Extracting archive
  - Installing masterminds/html5 (2.7.4): Extracting archive
  - Installing laminas/laminas-zendframework-bridge (1.1.1): Extracting archive
  - Installing laminas/laminas-stdlib (3.3.0): Extracting archive
  - Installing laminas/laminas-escaper (2.7.0): Extracting archive
  - Installing laminas/laminas-feed (2.13.0): Extracting archive
  - Installing psr/http-factory (1.0.1): Extracting archive
  - Installing laminas/laminas-diactoros (2.5.0): Extracting archive
  - Installing doctrine/lexer (1.2.1): Extracting archive
  - Installing egulias/email-validator (2.1.22): Extracting archive
  - Installing doctrine/annotations (1.11.1): Extracting archive
  - Installing doctrine/reflection (1.2.2): Extracting archive
  - Installing asm89/stack-cors (1.3.0): Extracting archive
  - Installing drupal/core (9.1.4): Extracting archive
  - Installing consolidation/comments (1.0.2): Extracting archive
  - Installing acquia/drupal-environment-detector (v1.3.0): Extracting archive
  - Applying patches for drupal/core
    https://www.drupal.org/files/issues/2020-02-07/2869592-remove-update-warning-34.patch (2869592 - Disabled update module shouldn't produce a status report warning)
    https://www.drupal.org/files/issues/2019-11-27/2815221-125.patch (2815221 - Add quickedit to the latest-revision route)
    https://www.drupal.org/files/issues/2020-12-15/3143958-6-subprofile-support-9.1.x.patch ([subprofile support] 1356276 - Allow profiles to define a base/parent profile and load them in the correct order / 2914389 - Allow profiles to exclude dependencies of their parent)
    https://www.drupal.org/files/issues/2020-04-01/2031261-118.patch (REMOVE: 2031261 - Fix SQLite variable limit)
    https://www.drupal.org/files/issues/2021-01-21/3143958-12-change-profile-config-sync-9.1.x.patch (REMOVE: Allow installation profile to be changed from Lightning during config sync)

  - Installing acquia/blt (12.7.0): Extracting archive
  - Installing oomphinc/composer-installers-extender (2.0.0): Extracting archive
  - Installing drupal/search_api (1.19.0): Extracting archive
  - Installing drupal/profile_switcher (1.0.0-alpha5): Extracting archive
  - Installing drupal/pendo (1.0.0-alpha4): Extracting archive
  - Installing drupal/jquery_ui (1.4.0): Extracting archive
  - Installing drupal/jquery_ui_draggable (1.2.0): Extracting archive
  - Installing drupal/jquery_ui_droppable (1.2.0): Extracting archive
  - Installing drupal/ctools (3.4.0): Extracting archive
  - Installing drupal/panels (4.6.0): Extracting archive
  - Installing drupal/panels_ipe (4.6.0)
  - Installing drupal/layout_library (1.0.0-beta2): Extracting archive
  - Installing drupal/ctools_block (3.4.0)
  - Installing drupal/core_context (1.0.0-beta4): Extracting archive
  - Installing drupal/panelizer (5.0.0-beta3): Extracting archive
  - Installing drupal/page_manager (4.0.0-beta6): Extracting archive
  - Installing drupal/moderation_sidebar (1.4.0): Extracting archive
  - Installing drupal/moderation_dashboard (1.0.0-beta2): Extracting archive
  - Installing phpdocumentor/reflection-common (2.2.0): Extracting archive
  - Installing phpdocumentor/type-resolver (1.4.0): Extracting archive
  - Installing phpdocumentor/reflection-docblock (5.2.2): Extracting archive
  - Installing drupal/token (1.9.0): Extracting archive
  - Installing drupal/redirect (1.6.0): Extracting archive
  - Installing drupal/pathauto (1.8.0): Extracting archive
  - Installing drupal/metatag (1.15.0): Extracting archive
  - Installing drupal/contact_storage (1.1.0): Extracting archive
  - Installing drupal/acquia_telemetry-acquia_telemetry (1.0.0-alpha5): Extracting archive
  - Installing drupal/lightning_core (5.5.0): Extracting archive
  - Installing ezyang/htmlpurifier (v4.13.0): Extracting archive
  - Installing caxy/php-htmldiff (v0.1.11): Extracting archive
  - Installing mkalkbrenner/php-htmldiff-advanced (0.0.8): Extracting archive
  - Installing drupal/diff (1.0.0): Extracting archive
  - Installing drupal/conflict (2.0.0-alpha2): Extracting archive
  - Installing drupal/autosave_form (1.2.0): Extracting archive
  - Installing drupal/lightning_workflow (3.16.0): Extracting archive
  - Installing vardot/blazy (1.8.2.1): Extracting archive
  - Installing npm-asset/dropzone (5.7.6): Extracting archive
  - Installing drupal/views_infinite_scroll (1.7.0): Extracting archive
  - Installing drupal/video_embed_field (2.4.0): Extracting archive
  - Installing drupal/blazy (2.1.0): Extracting archive
  - Installing drupal/slick (2.2.0): Extracting archive
  - Installing drupal/slick_entityreference (2.0.0): Extracting archive
  - Installing j7mbo/twitter-api-php (1.0.6): Extracting archive
  - Installing drupal/media_entity_twitter (2.6.0): Extracting archive
  - Installing drupal/media_entity_instagram (3.0.4): Extracting archive
  - Installing drupal/inline_entity_form (1.0.0-rc8): Extracting archive
  - Installing drupal/crop (2.1.0): Extracting archive
  - Installing drupal/image_widget_crop (2.3.0): Extracting archive
  - Installing drupal/embed (1.4.0): Extracting archive
  - Installing drupal/entity_embed (1.1.0): Extracting archive
  - Installing drupal/entity_browser (2.5.0): Extracting archive
  - Installing drupal/dropzonejs (2.4.0): Extracting archive
  - Installing bower-asset/jquery (3.5.1): Extracting archive
  - Installing bower-asset/slick-carousel (v1.8.1): Extracting archive
  - Installing bower-asset/cropper (v2.3.4): Extracting archive
  - Installing drupal/lightning_media (4.6.0): Extracting archive
  - Installing drupal/simple_gmap (3.0.0): Extracting archive
  - Installing drupal/layout_builder_styles (1.0.0-beta4): Extracting archive
  - Installing drupal/layout_builder_st (1.0.0-alpha2): Extracting archive
  - Installing drupal/layout_builder_restrictions (2.8.0): Extracting archive
  - Installing drupal/entity_browser_block (1.1.0): Extracting archive
  - Installing drupal/entity_block (1.0.0-beta3): Extracting archive
  - Installing drupal/bg_image_formatter (1.14.0): Extracting archive
  - Installing drupal/lightning_layout (2.9.0): Extracting archive
  - Installing league/event (2.2.0): Extracting archive
  - Installing lcobucci/jwt (3.4.5): Extracting archive
  - Installing paragonie/random_compat (v9.99.100): Extracting archive
  - Installing defuse/php-encryption (v2.2.1): Extracting archive
  - Installing league/oauth2-server (7.4.0): Extracting archive
  - Installing drupal/consumers (1.11.0): Extracting archive
  - Installing drupal/simple_oauth (4.5.0): Extracting archive
  - Installing swagger-api/swagger-ui (v3.43.0): Extracting archive
  - Installing drupal/openapi_ui (1.0.0-rc3): Extracting archive
  - Installing drupal/openapi_ui_swagger (1.0.0-rc4): Extracting archive
  - Installing drupal/openapi_ui_redoc (1.0.0-rc3): Extracting archive
  - Installing drupal/schemata (1.0.0-beta2): Extracting archive
  - Installing drupal/schemata_json_schema (1.0.0-beta2)
  - Installing drupal/openapi (2.0.0-rc3): Extracting archive
  - Installing drupal/openapi_rest (2.0.0-rc2): Extracting archive
  - Installing drupal/openapi_jsonapi (2.0.0): Extracting archive
  - Installing drupal/lightning_api (4.6.0): Extracting archive
  - Installing drupal/acquia_connector (2.0.0-rc4): Extracting archive
  - Installing seld/phar-utils (1.1.1): Extracting archive
  - Installing seld/jsonlint (1.8.3): Extracting archive
  - Installing react/promise (v2.8.0): Extracting archive
  - Installing justinrainbow/json-schema (5.2.10): Extracting archive
  - Installing composer/xdebug-handler (1.4.5): Extracting archive
  - Installing composer/spdx-licenses (1.5.5): Extracting archive
  - Installing composer/ca-bundle (1.2.9): Extracting archive
  - Installing composer/composer (2.0.11): Extracting archive
  - Installing acquia/lightning (5.1.2): Extracting archive
  - Installing drupal/core-recommended (9.1.4)
  - Installing drupal/mysql56 (1.1.0): Extracting archive
  - Applying patches for drupal/panels
    https://www.drupal.org/files/issues/panels-ipe-2878684-3.patch (2878684 - Use String.match to correlate regions when switching Layouts in Panels IPE)

  - Applying patches for drupal/panelizer
    https://www.drupal.org/files/issues/2020-03-23/2778565-47.patch (2778565 - Multilingual support for Panelizer)

51 package suggestions were added by new dependencies, use `composer suggest` to see details.
Package container-interop/container-interop is abandoned, you should avoid using it. Use psr/container instead.
Package doctrine/reflection is abandoned, you should avoid using it. Use roave/better-reflection instead.
Generating autoload files
51 packages you are using are looking for funding.
Use the `composer fund` command to find out more!
Scaffolding files for drupal/core:
  - Copy [project-root]/.editorconfig from assets/scaffold/files/editorconfig
  - Copy [project-root]/.gitattributes from assets/scaffold/files/gitattributes
  - Copy [web-root]/.csslintrc from assets/scaffold/files/csslintrc
  - Copy [web-root]/.eslintignore from assets/scaffold/files/eslintignore
  - Copy [web-root]/.eslintrc.json from assets/scaffold/files/eslintrc.json
  - Copy [web-root]/.ht.router.php from assets/scaffold/files/ht.router.php
  - Copy [web-root]/.htaccess from assets/scaffold/files/htaccess
  - Copy [web-root]/example.gitignore from assets/scaffold/files/example.gitignore
  - Copy [web-root]/index.php from assets/scaffold/files/index.php
  - Copy [web-root]/INSTALL.txt from assets/scaffold/files/drupal.INSTALL.txt
  - Copy [web-root]/README.txt from assets/scaffold/files/drupal.README.txt
  - Copy [web-root]/robots.txt from assets/scaffold/files/robots.txt
  - Copy [web-root]/update.php from assets/scaffold/files/update.php
  - Copy [web-root]/web.config from assets/scaffold/files/web.config
  - Copy [web-root]/sites/README.txt from assets/scaffold/files/sites.README.txt
  - Copy [web-root]/sites/development.services.yml from assets/scaffold/files/development.services.yml
  - Copy [web-root]/sites/example.settings.local.php from assets/scaffold/files/example.settings.local.php
  - Copy [web-root]/sites/example.sites.php from assets/scaffold/files/example.sites.php
  - Skip [web-root]/sites/default/default.services.yml: overridden in acquia/drupal-recommended-project
  - Skip [web-root]/sites/default/default.settings.php: overridden in acquia/drupal-recommended-project
  - Copy [web-root]/modules/README.txt from assets/scaffold/files/modules.README.txt
  - Copy [web-root]/profiles/README.txt from assets/scaffold/files/profiles.README.txt
  - Copy [web-root]/themes/README.txt from assets/scaffold/files/themes.README.txt
Scaffolding files for acquia/drupal-recommended-project:
  - Copy [web-root]/sites/default/default.services.yml from docroot/core/assets/scaffold/files/default.services.yml
  - Copy [web-root]/sites/default/default.settings.php from docroot/core/assets/scaffold/files/default.settings.php
Creating BLT template files...
 > /home/ide/project/vendor/acquia/blt/bin/blt internal:add-to-project --ansi -n
> blt:init:settings
Could not find blt/example.local.blt.yml/. Create and commit this file if you'd like to automatically generate blt/local.blt.yml/ based on this template.
Generating hash salt...
[File\Write] Writing to /home/ide/project/salt.txt.
> blt:init:shell-alias
The BLT alias is already installed and up to date.
[Acquia\Blt\Robo\Tasks\GitTask] Running git commands...
[Acquia\Blt\Robo\Tasks\GitTask] git init && git commit -m 'Initial commit.' --allow-empty && git add -A && git commit -m 'Created BLT project.'
[Acquia\Blt\Robo\Tasks\GitTask]  Initialized empty Git repository in /home/ide/project/.git/ 
[Acquia\Blt\Robo\Tasks\GitTask]  Exit code 128 
[error]  Could not initialize new git repository.
For troubleshooting guidance and support, see https://docs.acquia.com/blt/support/ 
BLT installation failed! Please execute /home/ide/project/vendor/acquia/blt/bin/blt internal:add-to-project --ansi -n --verbose to debug the issue.

                                     
  [Exception]                        
  Installation aborted due to error  
                                     

update [--with WITH] [--prefer-source] [--prefer-dist] [--dry-run] [--dev] [--no-dev] [--lock] [--no-install] [--no-autoloader] [--no-scripts] [--no-suggest] [--no-progress] [-w|--with-dependencies] [-W|--with-all-dependencies] [-v|vv|vvv|--verbose] [-o|--optimize-autoloader] [-a|--classmap-authoritative] [--apcu-autoloader] [--apcu-autoloader-prefix APCU-AUTOLOADER-PREFIX] [--ignore-platform-req IGNORE-PLATFORM-REQ] [--ignore-platform-reqs] [--prefer-stable] [--prefer-lowest] [-i|--interactive] [--root-reqs] [--] [<packages>]...

[notice] Command: 'composer' 'update' [Exit: 1]
Reinitialized existing Git repository in /home/ide/project/.git/
[notice] Command: 'git' 'init' [Exit: 0]
[notice] Command: 'git' 'add' '-A' [Exit: 0]
Author identity unknown

*** Please tell me who you are.

Run

  git config --global user.email "you@example.com"
  git config --global user.name "Your Name"

to set your account's default identity.
Omit --global to set the identity only in this repository.

fatal: unable to auto-detect email address (got 'ide@a3fefbcd021b.(none)')
[notice] Command: 'git' 'commit' '--message' 'Initial commit.' '--quiet' [Exit: 128]

New 💧Drupal project created in /home/ide/project. 🎉
```

**Comment by danepowell** [comment]
You can resolve this by setting your global Git user email and name. I thought Cloud IDEs already did this, do they not?

I also thought BLT set a dummy Git email and name if they are absent: https://github.com/acquia/blt/pull/3541

**Comment by anavarre** [comment]
Is it really failing because of that? It didn't occur to me it would.

Also yes, Cloud IDE does set the Git email and username but I was testing in a local env without being authenticated against the API, thus why it wasn't set.

**Comment by danepowell** [comment]
@anavarre I cannot reproduce this, can you provide steps to reproduce, including an environment configuration that produces the problem?

I tried clearing my git username and email locally and running the `acli new` command and it worked fine, which is the behavior I'd expect given that BLT validates Git config before running.

You can see BLT's logic that validates Git config here: https://github.com/acquia/blt/blob/dfeb41727fcf226041ba868994b6075fbd3dd324/src/Robo/Hooks/ValidateHook.php#L106

**Comment by anavarre** [comment]
@danepowell to reproduce, create a new IDE, confirm the Git config returns the following and install the recommended project with `acli new`

```
$ git config -l
alias.co=checkout
alias.br=branch
alias.ci=commit
alias.st=status
alias.unstage=reset HEAD --
alias.last=log -1 HEAD
```

We have logic to backfill the Git username and email address, but when it hasn't happened yet (that is, when you're not yet authenticated against the Cloud Platform API) then we will not have been able to take care of this if what you begin your journey with is `acli new`.

I very much understand https://github.com/acquia/blt/blob/main/src/Robo/Hooks/ValidateHook.php#L106 should kick in, but apparently it's not :shrug:

**Comment by danepowell** [comment]
Thanks, I was able to reproduce following those steps and confirm it's a BLT bug: https://github.com/acquia/blt/pull/4357

**Comment by danepowell** [comment]
@anavarre this is fixed in BLT 12.8.1. However it looks like Acquia CLI _also_ runs a Git commit after initializing the project, and this fails for the same reason.

It seems like either BLT should stop doing a Git commit, or Acquia CLI shouldn't initialize a Git directory if it already exists. Do you have a preference?

**Comment by anavarre** [comment]
My immediate thinking is I wouldn't want any tool to `git commit` for me. However I can see how that can be helpful to get started, especially when you don't know what you're doing.

Apart from that, checking that a Git dir exists before you initialize it looks the right course of action, no?
